opencensus/metrics_listener.go

```diff
@@ -0,0 +1,48 @@
+package opencensus
+
+import (
+	"time"
+
+	"go.opencensus.io/stats"
+
+	gosundheit "github.com/AppsFlyer/go-sundheit"
+)
+
+// MetricsListener reports metrics on each check registration, start and completion event (as gosundheit.CheckListener)
+// This listener all reports metrics for the entire service health (as gosundheit.HealthListener)
+type MetricsListener struct {
+	classification string
+}
+
+func NewMetricsListener(opts ...Option) *MetricsListener {
+	listener := &MetricsListener{}
+
+	for _, opt := range append(opts, WithDefaults()) {
+		opt(listener)
+	}
+
+	return listener
+}
+
+func (c *MetricsListener) OnCheckRegistered(name string, result gosundheit.Result) {
+	c.recordCheck(name, result)
+}
+
+func (c *MetricsListener) OnCheckStarted(_ string) {
+}
+
+func (c *MetricsListener) OnCheckCompleted(name string, result gosundheit.Result) {
+	c.recordCheck(name, result)
+}
+
+func (c *MetricsListener) OnResultsUpdated(results map[string]gosundheit.Result) {
+	allHealthy := allHealthy(results)
+	allChecksCtx := createMonitoringCtx(c.classification, ValAllChecks, allHealthy)
```

Comment:
Should probably be a separate PR, but it's worth noting that all checks/all healthy should be an entirely different stat than the individual checks. I'd at least add a todo here.

Reply:
This preserve the original report that exported by the lib. We should _discuss_ if we wish to change it in some other issue.

Reply:
It was a mistake but fixing it will be a breaking change, and I know our dev are using it.
I'd defer this change to later